pbft/pbft_checkpoint_manager.cpp

```diff
@@ -0,0 +1,323 @@
+// Copyright (C) 2018 Bluzelle
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License, version 3,
+// as published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+#include <pbft/pbft_checkpoint_manager.hpp>
+#include <proto/pbft.pb.h>
+#include <utils/bytes_to_debug_string.hpp>
+#include <utils/make_endpoint.hpp>
+#include <algorithm>
+#include <pbft/pbft.hpp>
+#include <iterator>
+
+namespace
```

Comment:
I've read through this module and I think it is mostly extracted from pbft. So the functionality must be good. 

The code formatting looks very nice as well. I think the re factoring is good.